egs/librispeech/ASR/conformer_ctc/conv1d_abs_attention.py

```diff
@@ -0,0 +1,211 @@
+# coding=utf-8
+import math
+import collections
+from itertools import repeat
+
+import torch
+from torch import Tensor
+from torch.nn.parameter import Parameter
+from torch.nn import functional as F
+from torch.nn import init
+from torch.nn.modules.module import Module
+from torch.nn.common_types import _size_1_t
+from typing import Optional, Tuple
+
+
+def _ntuple(n):
```

Comment:
Could you put a note here about where the original code comes from?

Reply:
OK.